canonical-webteam/generator-canonical-webteam
# Intelligently update blocks in gitignore

If a `.gitignore` file exists in the project, compare the blocks in the
file with the gitignore template and if blocks with the same IDs exist,
then replace them.

QA
--

``` bash
sudo npm link   # Use this version of generator-canonical-webteam
git clone https://github.com/canonical-websites/tour.ubuntu.com  # Experiment with tour
cd tour.ubuntu.com
yo canonical-webteam:run
```

Now answer `Y` to replacing the `.gitignore` file. Inspect the new file, and see all the new blocks from https://gist.github.com/nottrobin/38a484926513173e6ec0511999f07f6c are there, but that the old ignore rules remain below.

Now edit `.gitignore`, and change one of the new blocks.

Run `yo canonical-webteam:run` again to replace the .gitignore again. See that your changes are replaced.
